ivy_tests/test_ivy/test_functional/test_core/test_creation.py

```diff
@@ -2,159 +2,358 @@
-        ivy.functional.backends.numpy.logspace(
-            ivy.to_numpy(start), ivy.to_numpy(stop), num, base, axis, device=device
-        ),
+def test_logspace(
+    start,
+    stop,
+    num,
+    base,
+    axis,
+    dtype,
+    as_variable,
+    with_out,
+    num_positional_args,
+    native_array,
+    container,
+    instance_method,
+    fw,
+):
+    helpers.test_array_function(
+        dtype,
+        as_variable,
+        with_out,
+        num_positional_args,
+        native_array,
+        container,
+        instance_method,
+        fw,
+        "logspace",
+        start = start,
+        stop = stop,
+        num = num,
+        base = base,
+        axis = axis,
     )
 
 
-# Still to Add #
-# ---------------#
-
 # arange()
+@given(
+    start = st.integers(min_value = 0), 
+    stop = st.integers(min_value = 0) | st.none(),
+    step = st.integers().filter(lambda x : True if x != 0 else False),
+    dtype = st.sampled_from(ivy_np.valid_int_dtypes),
+    as_variable=st.booleans(),
+    with_out=st.booleans(),
+    num_positional_args=helpers.num_positional_args(fn_name="arange"),
+    native_array=st.booleans(),
+    container=st.booleans(),
+    instance_method=st.booleans(),
+)
+def test_arange(
+    start,
+    stop,
+    step,
+    dtype,
+    as_variable,
+    with_out,
+    num_positional_args,
+    native_array,
+    container,
+    instance_method,
+    fw,
+):
+    helpers.test_array_function(
+        dtype,
+        as_variable,
+        with_out,
+        num_positional_args,
+        native_array,
+        container,
+        instance_method,
+        fw,
+        "arange",
+        start = start,
+        stop = stop,
+        step = step,
+    )
+
+
 # asarray()
+@given(
+    dtype_and_x=helpers.dtype_and_values(ivy_np.valid_int_dtypes),
+    as_variable=st.booleans(),
+    with_out=st.booleans(),
+    num_positional_args=helpers.num_positional_args(fn_name="asarray"),
+    native_array=st.booleans(),
+    container=st.booleans(),
+    instance_method=st.booleans(),
+)
+def test_asarray(
+    dtype_and_x,
+    as_variable,
+    with_out,
+    num_positional_args,
+    native_array,
+    container,
+    instance_method,
+    fw,
+):
+    dtype, x = dtype_and_x
+    helpers.test_array_function(
+        dtype,
+        as_variable,
+        with_out,
+        num_positional_args,
+        native_array,
+        container,
+        instance_method,
+        fw,
+        "asarray",
+        object_in = np.asarray(x),
+    )
+
+
 # empty()
+@given(
+    array_shape = st.integers(0,5) | st.lists(st.integers(min_value = 0, max_value=10), min_size = 1, max_size = 5),
+    dtype = st.sampled_from(ivy_np.valid_int_dtypes),
+    as_variable=st.booleans(),
+    with_out=st.booleans(),
+    num_positional_args=helpers.num_positional_args(fn_name="empty"),
+    native_array=st.booleans(),
+    container=st.booleans(),
+    instance_method=st.booleans(),
```

Comment:
`instance_method` is not relevant here as the input contains no arrays. We should remove from the decorator and hard-code it as `False` in the inputs to `helpers.test_array_function`